ee/desktop/runner/runner_darwin.go

```diff
@@ -0,0 +1,124 @@
+//go:build darwin
+// +build darwin
+
+package runner
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/go-kit/kit/log/level"
+)
+
+func (r *DesktopUsersProcessesRunner) consoleUsers() ([]string, error) {
+	consoleOwnerUid, err := consoleOwnerUid()
+	if err != nil {
+		return nil, fmt.Errorf("getting console owner uid: %w", err)
+	}
+
+	if consoleOwnerUid == "" {
+		return []string{}, nil
+	}
+
+	// convert string to int
+	uid, err := strconv.ParseInt(consoleOwnerUid, 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("converting uid to int: %w", err)
+	}
+
+	// there seems to be a brief moment during start up where root or system (non-human)
+	// users own the console, if we spin up the process for them it will add an
+	// unnecessary process. On macOS human users start at 501
+	if uid < 501 {
+		level.Debug(r.logger).Log(
+			"msg", "skipping desktop for root or system user",
+			"uid", consoleOwnerUid,
+		)
+
+		return []string{}, nil
+	}
+
+	return []string{consoleOwnerUid}, nil
+}
+
+func consoleOwnerUid() (string, error) {
+	const cmd = `echo "show State:/Users/ConsoleUser" | scutil`
+	output, err := exec.Command("sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("getting console user: %w", err)
+	}
```

Comment:
Several comments

1. You shouldn't use an `sh` exec. 
2. Instead of `echo ...` just set stdin.
3. You should use a exec.CommandContext with timeout

Reply:
updated ... much better